### PR TITLE
[SAI] Move constructor to sai_api_initialize

### PR DESCRIPTION
**Pull request by kcudnik**
For future passing parameters via service method table

@lguohan @marian-pritsak @Pterosaur @chrispsommers please review

### CODE REVIEW

File: dash-pipeline/tests/libsai/vnet_out/vnet_out.cpp [hunk 1]
```diff
@@ -3,8 +3,13 @@
 #include <unordered_map>
 #include <string.h>
 
+extern "C" {
 #include <sai.h>
+}
 
+// TODO below apis are extern of internal implementation
+// they should not be used by extern, but obtained via
+// sai_api_query function, FIXME
```
Comment (chrispsommers) [inline dash-pipeline/tests/libsai/vnet_out/vnet_out.cpp:12]: Could you add an issue to ensure tracking? Thanks.

Comment (kcudnik) [reply dash-pipeline/tests/libsai/vnet_out/vnet_out.cpp:12]: sure

Comment (kcudnik) [reply dash-pipeline/tests/libsai/vnet_out/vnet_out.cpp:12]: done https://github.com/sonic-net/DASH/issues/414

**Review by Pterosaur — APPROVED** [review]
lgtm

**Comment by kcudnik** [comment]
please merge if is ok, i dont have permissions here

**Pull request merged**